01_financial-well-being/income_metrics_all.qmd

```diff
@@ -0,0 +1,676 @@
+                                     size_flag==0 & puma_flag==3 ~ 3,
+                                     size_flag==1 ~ 3)
+  )
+
+```
+
+Keep only relevant variables
+
+```{r}
+
+metrics_income <- metrics_income %>% 
+  select(statefip, place, pctl_20, pctl_20_quality, 
+         pctl_50, pctl_50_quality, pctl_80, 
+         pctl_80_quality, size_flag, 
+         puma_flag)
+
+```
+
+### Evaluate results
+
+Look at distributions of income percentiles by place.
+
+```{r}
+
+metrics_income %>% 
+  ungroup() %>% 
+  select(place, pctl_20, pctl_50, pctl_80) %>% 
+  pivot_longer(cols = c(pctl_20, pctl_50, pctl_80), names_to = "income", values_to = "percentile") %>% 
+  ggplot(aes(x = percentile, color = income, fill = income)) +
+  geom_density(alpha = 0.15) +
+  theme_minimal() +
+  ggtitle("Income percentiles") +
+  ylab("Density")
+  
+```
+
+## Export {#export}
+
+Rename state.
+
+```{r}
+metrics_income <- metrics_income %>% 
+  rename('state' = 'statefip')
+```
+
+Add a variable for the year of the data
+
+```{r}
+
+metrics_income <- metrics_income %>%
+  mutate(
+    year = 2021
+  )
+
+
+```
+
+Order the variables how we want
+
+```{r}
+
+metrics_income <- metrics_income %>% 
+  select(year, state, place, pctl_20, pctl_20_quality, pctl_50, 
+         pctl_50_quality, pctl_80, pctl_80_quality)
+
+```
+
+Export as CSV
+
+```{r}
+
+write_csv(metrics_income, "data/final/metrics_income_city_2021.csv")
+
+```
+
+## Compare with counties
+
+Compare results from places that share boundaries with counties against the results of those counties. These should be the same.
+
+First read in the FIPs information for these places (CSV created from a file put together by Aaron Williams).
+
+```{r}
+
+independent_cities <- read_csv("data/raw/independent_cities.csv") %>% 
+    mutate(state = sprintf("%0.2d", as.numeric(state)),
+         county = sprintf("%0.3d", as.numeric(county)),
+         place = sprintf("%0.5d", as.numeric(place))) %>% 
+  mutate(state = as.character(state))
+```
+
+Net read in the final place and county data.
+
+```{r}
+metrics_income_place <- read_csv("data/final/metrics_income_city_2021.csv") %>% 
+  left_join(independent_cities, by = c("state" = "state", "place" = "place")) %>% 
+  filter(!is.na(county))
+
+```
+
+
+Read in county and select just needed variables.
+```{r}
+
+metrics_income_county <- read_csv("data/final/metrics_income_county_2021.csv") %>% 
+  select(state, county, pctl_20_county = pctl_20,
+         pctl_50_county = pctl_50,
+         pctl_80_county = pctl_80
+         )
+
+```
+
+Join onto place data and calculate difference between income metrics. No difference. 
+```{r}
+
+metrics_income_place %>% 
+  left_join(metrics_income_county, by = c("state" = "state", "county" = "county")) %>%mutate(diff_20 = pctl_20 - pctl_20_county,
+         diff_50 = pctl_50 - pctl_50_county,
+         diff_80 = pctl_80 - pctl_80_county) %>% 
+  select(state, place, county, diff_20, diff_50, diff_80) %>% 
+  reactable(
+    
+  )
```

Comment:
I created a Quarto doc that does this for all metrics. It's on branch `fix_missing`.